### ISSUE TITLE
Missing Dividend Data

### ISSUE THREAD

**Issue by andrewlindemulder** [polygon-io/issues#288]
**URL**
https://api.polygon.io/v3/reference/dividends?ticker=QQQY&limit=1&sort=ex_dividend_date&apiKey=***************

**Result**
`{
    "results": [
        {
            "cash_amount": 0.85,
            "currency": "USD",
            "declaration_date": "2024-01-30",
            "dividend_type": "CD",
            "ex_dividend_date": "2024-02-01",
            "frequency": 12,
            "pay_date": "2024-02-06",
            "record_date": "2024-02-02",
            "ticker": "QQQY"
        }
    ],
    "status": "OK",
    "request_id": "8e2cd56ab71283aa47b38f3f59d13bbd",
    "next_url": "https://api.polygon.io/v3/reference/dividends?cursor=YXA9MjAyMy0xMi0yOCZhcz1RUVFZJmxpbWl0PTEmb3JkZXI9ZGVzYyZzb3J0PWV4X2RpdmlkZW5kX2RhdGUmdGlja2VyPVFRUVk"
}`

**Expected Result**
QQQY paid their most recent dividend on 3/6/2024 with an ex-dividend date of 3/1/2024. 12 days past ex-dividend and the data is not available yet.

**Screenshots**
![image](https://github.com/polygon-io/issues/assets/55672860/2704b800-2a41-4e6e-9122-66bda825b93f)
![image](https://github.com/polygon-io/issues/assets/55672860/730d2688-174c-493c-b252-030a06618bba)

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04
 - Browser: Chrome
 - Version: 122.0.6261.112 
 
**Additional context**
This is repeatable with other tickers as well, QQQY, SDIV, IWMY, JEPQ, etc.

**Comment by andrewlindemulder** [comment]
I see it populating now, but the declaration dates are wrong. Declaration dates are showing 2024-01-09 but the actual declaration date is 2024-02-29 for IWMY and QQQY.

**Comment by andrewlindemulder** [comment]
Dividend data is not correct and untrustworthy for many tickers, beyond this issue there is serious delay between when dividends are declared, paid, etc. before they are available in the feed, sometimes to the tune of 2 weeks +. If these issues cannot be resolved, i cannot continue with Polygon as the data is not reliable.. example below, in the second example USOI has never paid a dividend as high $8.48:
![image](https://github.com/polygon-io/issues/assets/55672860/d4cd79b4-1f4d-47c5-9952-5f7aacdea324)
![image](https://github.com/polygon-io/issues/assets/55672860/82f991a5-5b0c-4321-ae8f-0a510dc8370e)

**Comment by andrewlindemulder** [comment]
@AHangstefer are you able to review this please and provide an update? I have opened a ticket with support, and am not getting a response, i messaged via the online chat feature on your website and have not gotten a response, i've opened 2 github issues and am not getting a response.

**Comment by andrewlindemulder** [comment]
Another data point: EAGL snapshot shows the below for today.

`{"day": {"low": 25.11, "otc": null, "high": 25.15, "open": 25.11, "vwap": 25.1356, "close": 25.14, "volume": 34440, "timestamp": null, "transactions": null}, "min": {"low": 25.14, "otc": null, "high": 25.14, "open": 25.14, "vwap": 25.14, "close": 25.14, "volume": 200, "timestamp": 1711392780000, "transactions": 1, "accumulated_volume": 34440}, "ticker": "EAGL", "prevDay": {"low": 10.01, "otc": null, "high": 10.2, "open": 10.19, "vwap": 10.113, "close": 10.1, "volume": 42661, "timestamp": null, "transactions": null}, "updated": 1711392840000000000, "last_quote": null, "last_trade": null, "todays_change": 15.04, "fair_market_value": null, "todays_change_percent": 148.91089108910893}`

However yesterday's close for EAGL was $25.11, not $10.1

**Comment by andrewlindemulder** [comment]
@qrpike I am concerned about the accuracy of the data I am receiving from Polygon, and I have not seen a response from support in any channel.

**Comment by darinfisher** [comment]
I am also seeing this issue with a few including KSS, THTA, PBR.  Most are missing data and the latest for PBR is just wrong.

**Comment by andrewlindemulder** [comment]
Any update on this?

**Comment by ViewsTiger** [comment]
This is an issue that I'm experiencing as well, even on well-known tickers. I run this query every day, and my app hasn't pulled any new dividends since 6/26.
For example, DuPont (DD) announced a dividend on 6/27, yet Polygon still shows the latest dividend announcement from them as 4/17. 
Pulling dividend dates for my portfolio and watchlist is a key part of my app and the primary reason I chose to use Polygon over other data providers. 
Is this issue still ongoing for anyone else?

There have been tons of new dividend announcements since then, as noted here (https://www.marketbeat.com/dividends/announcements/), of the few I sampled, none of them were updated in Polygon

**Comment by darinfisher** [comment]
This issue has become worse.  Now much of the history for each equity is missing and I am missing updates for many.

**Comment by andrewlindemulder** [comment]
@qrpike @AHangstefer any update on this? Can you let us know if you are looking into it at all?

**Comment by qrpike** [comment]
@andrewlindemulder Yes, we have identified the underlying issue and are actively working on the fix.

**Comment by ViewsTiger** [comment]
It seems to be working now, my app is pulling down a bunch of new dividends

**Comment by qrpike** [comment]
@ViewsTiger Yes, we have fixed the issue. We are looking at refactoring this service since it has proved brittle in this scenario. 

CC: @andrewlindemulder @darinfisher
